paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -246,19 +246,24 @@ def get_authors_to_be_notified(self):
         else:
             return f"(Could not get authors: {authors})"
 
-    def deploy_group_is_set_to_notify(self):
+    def deploy_group_is_set_to_notify(self, notify_type):
         for step in self.deploy_info.get('pipeline', []):
             if step.get('step', '') == self.deploy_group:
-                return step.get('slack_notify', False)
+                # If specific notify_type is turned on, then return True
+                if step.get(notify_type, False):
+                    return True
+                # If the specific notify_type isn't given, it defaults to True.
+                return step.get('slack_notify', False) and step.get(notify_type, True)
         return False
 
     def post(self, **kwargs):
-        if self.deploy_group_is_set_to_notify():
-            self.sc.post(**kwargs)
-        else:
-            log.debug(f"{self.deploy_group} isn't set to notify slack")
+        self.sc.post(**kwargs)
 
     def notify_after_mark(self, ret):
+        if not self.deploy_group_is_set_to_notify('notify_after_mark'):
+            log.debug(f"{self.deploy_group}.notify_after_mark isn't set to notify slack")
+            return
```

Comment:
I think I'd prefer to keep this if statement inside `post`, and have `post` take notify_type as an argument.

Reply:
Done